src/ingest_validation_tools/upload.py

```diff
@@ -718,228 +672,335 @@ def _get_constraint_check_errors(
-            field = f"{ref}_path"
-            if not row.get(field):
                 continue
-            unique_paths.add(row[field])
-        if ref == OtherTypes.CONTRIBUTORS:
-            schema.contributors_paths = [
-                str(Path(Path(metadata_path).parent, path)) for path in unique_paths
-            ]
-        elif ref == OtherTypes.ANTIBODIES:
-            schema.antibodies_paths = [
-                str(Path(Path(metadata_path).parent, path)) for path in unique_paths
-            ]
-        for path_value in sorted(unique_paths):
-            self._check_path(path_value, ref, schema, metadata_path)
-
-    def _check_data_path(
-        self, schema_version: SchemaVersion, metadata_path: Path, path_value: str
-    ) -> Dict[str, List[str]]:
-        errors = {}
-        data_path = Path(path_value)
-        print_path = str(Path(self.directory_path / data_path))
-
-        if not schema_version.dir_schema:
-            raise Exception(
-                f"No directory schema found for data_path " f"{print_path} in {metadata_path}!"
-            )
-
-        try:
-            ref_errors = get_data_dir_errors(
-                schema_version.dir_schema,
-                root_path=self.directory_path,
-                data_dir_path=data_path,
-                dataset_ignore_globs=self.dataset_ignore_globs,
-            ).popitem()
-            if type(ref_errors[1]) is list:
-                errors[f"{print_path} (as {ref_errors[0]})"] = ref_errors[1]
+            # After this point dir can be assumed to exist, use actual path (print_path) for logging
+            except Exception as e:
+                self.errors.directory[print_path] = e
+                continue
+            if ref_errors[1]:
+                self.errors.directory[f"{print_path} (as {ref_errors[0]})"] = ref_errors[1]
+            # TODO: Different dataset dirs within an upload could validate against
+            # different minor versions. Is this desirable? Logging as a curiosity for now,
+            # still only capturing final version.
+            print(f"Dir schema {ref_errors[0]} used for {print_path}")
             schema_version.dir_schema = ref_errors[0]
-        except FileNotFoundError:
-            self.errors.directory[str(metadata_path)].append(
-                f"Value '{path_value}' in column 'data_path' points to non-existent directory: '{self.directory_path / path_value}'"
-            )
-        except Exception as e:
-            errors[print_path] = e
-        return errors
 
-    def _check_other_path(self, other_path: str):
-        schema = get_schema_version(
-            Path(other_path),
-            self.encoding,
-            self.app_context["entities_url"],
-            self.app_context["ingest_url"],
-            self.globus_token,
-            self.directory_path,
-        )
-        self.validation_routine(tsv_paths={str(other_path): schema})
+    def get_error_path(self, data_path: Path) -> str:
```

Comment:
Allows for shared upload-aware error reporting

NIHDEV-803 global/non_global error reporting uses `data_path` value in error